.github/workflows/tests.yml

```
@@ -41,8 +48,8 @@ jobs:
       pytest_store_durations: false
 
   upload-stable-deps:
-    needs: tests
-    runs-on: ubuntu-latest
+    needs: [tests, determine_runner]
+    runs-on: ${{ needs.determine_runner.outputs.runner_group }}
     if: ${{ github.event_name == 'push' }}

```

This step should only be run on schedule. I think we missed that when we merged #6164. Since it only runs on schedule, it might not be needed to run this step on the large runners. In that case, we don't need the `determine-runner` step in this file.

```suggestion
    needs: [tests, determine_runner]
    runs-on: ${{ needs.determine_runner.outputs.runner_group }}
    if: ${{ github.event_name == 'schedule' }}

```

---

You are right, I will fix it up